src/models/BibDetails.ts

```diff
@@ -0,0 +1,202 @@
+import type {
+  LinkedBibDetail,
+  BibDetail,
+  FieldMapping,
+} from "../types/bibTypes"
+import { preProcess } from "../utils/bibModelPreprocessing"
+
+export default class BibDetailsModel {
+  bib: ProcessedSearchResult
+
+  constructor(bib: SearchResult) {
+    // Preprocessing includes grouping notes into an object with labels as keys,
+    // as well as interleaving parallels into primary fields
+    this.bib = preProcess(bib)
+    // this.subjectLiteral = "to do investigate  subject heading api fallback scenario"
+  }
+  get holdingsDetails() {
+    const holdings = this.bib.holdings
+    if (!holdings) return []
+    return [
+      { label: "Location", field: "location" },
+      { label: "Format", field: "format" },
+      { label: "Call Number", field: "shelfMark" },
+      { label: "Library Has", field: "holdingStatement" },
+      { label: "Notes", field: "notes" },
+    ]
+      .map((fieldMapping) => {
+        const detail = this.buildHoldingDetail(fieldMapping)
+        return detail
+      })
+      .filter((f) => f)
+  }
+  get topDetails() {
+    return [
+      { field: "titleDisplay", label: "Title" },
+      { field: "publicationStatement", label: "Published By" },
+      // external link
+      { field: "supplementaryContent", label: "Supplementary Content" },
+      // internal link
+      { field: "creatorLiteral", label: "Author" },
+    ]
+      .map((fieldMapping) => {
+        let detail
+        if (fieldMapping.field === "supplementaryContent")
+          detail = this.supplementaryContent
+        else if (fieldMapping.field === "creatorLiteral")
+          detail = this.buildInternalLinkedDetail(fieldMapping)
+        else detail = this.buildStandardDetail(fieldMapping)
+        if (!detail) return
+        return detail
+      })
+      .filter((f) => f)
+  }
+  get bottomDetails() {
+    return [
+      { field: "contributorLiteral", label: "Additional Authors" },
+      { field: "partOf", label: "Found In" },
+      { field: "serialPublicationDates", label: "Publication Date" },
+      { field: "extent", label: "Description" },
+      { field: "description", label: "Summary" },
+      { field: "donor", label: "Donor/Sponsor" },
+      { field: "seriesStatement", label: "Series Statement" },
+      { field: "uniformTitle", label: "Uniform Title" },
+      { field: "titleAlt", label: "Alternative Title" },
+      { field: "formerTitle", label: "Former Title" },
+      { field: "subjectLiteral", label: "Subjects" },
+      { field: "genreForm", label: "Genre/Form" },
+      { field: "notes", label: "Notes" },
+      { field: "tableOfContents", label: "Contents" },
+      { field: "shelfMark", label: "Call Number" },
+      { field: "isbn", label: "ISBN" },
+      { field: "issn", label: "ISSN" },
+      { field: "oclc", label: "OCLC" },
+      { field: "lccn", label: "LCCN" },
+    ]
+      .map((fieldMapping) => {
+        let detail
```

Comment:
I believe you don't need to instantiate and assign to a variable, you can just return from within the if statements.